SpatialGDK/Source/SpatialGDKTests/Examples/SpatialGDKExampleTest.cpp

```diff
@@ -0,0 +1,172 @@
+
+	if (TimePassed >= MIN_WAIT_TIME_FOR_BACKGROUND_COMPUTATION)
+	{
+		FScopeTryLock BackgroundComputationLock(&ComputationResult.Mutex);
+
+		if (BackgroundComputationLock.IsLocked())
+		{
+			Test->TestTrue("Computation result is equal to expected value", ComputationResult.Value == 42);
+			return true;
+		}
+
+		if (TimePassed >= MAX_WAIT_TIME_FOR_BACKGROUND_COMPUTATION)
+		{
+			Test->TestTrue("Computation finished in time", false);
+			return true;
+		}
+	}
+
+	return false;
+}
+
+EXAMPLE_SIMPLE_TEST(GIVEN_initial_value_WHEN_performing_background_compuation_THEN_the_result_is_correct)
+{
+	ADD_LATENT_AUTOMATION_COMMAND(FStartBackgroundThreadComputation());
+	ADD_LATENT_AUTOMATION_COMMAND(FWaitForComputationAndCheckResult(this));
+
+	return true;
+}
+
+// 2. Simple test example
+EXAMPLE_SIMPLE_TEST(GIVEN_one_and_two_WHEN_summed_THEN_the_sum_is_three)
+{
+	int X = 1;
+	int Y = 2;
+	int Sum = X + Y;
+
+	TestTrue("The sum is correct", Sum == 3);
+
+	return true;
+}
+
+// 3. Complex test example
+EXAMPLE_COMPLEX_TEST(ComplexTest);
+bool ComplexTest::RunTest(const FString& Parameters)
+{
+	TArray<FString> OutArray;
+	Parameters.ParseIntoArrayWS(OutArray);
+
+	if (OutArray.Num() != 3)
+	{
+		UE_LOG(LogSpatialGDKExamples, Error, TEXT("Invalid Test Input"));
+		return true;
+	}
+
+	TArray<int> ArgsAndResult;
+	for (const auto& Value : OutArray)
+	{
+		if (Value.IsNumeric())
+		{
+			ArgsAndResult.Push(FCString::Atoi(*Value));
+		}
+	}
+
+	int Sum = ArgsAndResult[0] + ArgsAndResult[1];
+	TestTrue("The sum is correct", Sum == ArgsAndResult[2]);
+
+	return true;
+}
+
+void ComplexTest::GetTests(TArray<FString>& OutBeautifiedNames, TArray <FString>& OutTestCommands) const
+{
+	OutBeautifiedNames.Add(TEXT("GIVEN_two_and_two_WHEN_summed_THEN_the_sum_is_four"));
+	OutTestCommands.Add(TEXT("2 2 4"));
+
+	OutBeautifiedNames.Add(TEXT("GIVEN_three_and_five_WHEN_summed_THEN_the_sum_is_eight"));
+	OutTestCommands.Add(TEXT("3 5 8"));
+}
+
+// 4. Example of test fixture
+namespace
+{
+const FString ExampleTestFolder = FPaths::Combine(FPaths::ProjectContentDir(), TEXT("ExampleTests/"));
+
+class ExampleTestFixture
+{
+public:
+	ExampleTestFixture()
+	{
+		CreateTestFolders();
+	}
+	~ExampleTestFixture()
+	{
+		DeleteTestFolders();
+	}
+
+private:
+
+	void CreateTestFolders()
+	{
+		IPlatformFile& PlatformFile = FPlatformFileManager::Get().GetPlatformFile();
+		PlatformFile.CreateDirectoryTree(*ExampleTestFolder);
+	}
+
+	void DeleteTestFolders()
+	{
+		IPlatformFile& PlatformFile = FPlatformFileManager::Get().GetPlatformFile();
+		PlatformFile.DeleteDirectoryRecursively(*ExampleTestFolder);
+	}
+};
+} // anonymous namespace
+
+EXAMPLE_SIMPLE_TEST(GIVEN_empty_folder_WHEN_creating_a_file_THEN_the_file_has_been_created)
+{
+	ExampleTestFixture Fixture;
+
+	FString FilePath = FPaths::Combine(ExampleTestFolder, TEXT("Example.txt"));
+
+	IPlatformFile& PlatformFile = FPlatformFileManager::Get().GetPlatformFile();
+	IFileHandle* Handle = PlatformFile.OpenWrite(*FilePath);
```

Comment:
Use unique_ptr

Reply:
I'd use unique_ptr, but as far as I can tell, that's a UE4 convention to use raw pointers for such cases?

Reply:
Do we know why that is a convention? Are we sure it's not just that it's old code? This is literally the kind of thing that unique ptr is meant to solve, so if unreal doesn't it might not be a convention we want to follow.

Reply:
Changed to TUniquePtr